sql/mysqld.cc

```diff
@@ -7488,42 +7479,7 @@ static int init_server_components() {
       if (binlog_space_limit)
         LogErr(WARNING_LEVEL, ER_NEED_LOG_BIN, "--binlog-space-limit");
     }
-<<<<<<< HEAD
 #ifdef WITH_WSREP
-||||||| 35582423e36
-  } else if (expire_logs_days_supplied)
-    binlog_expire_logs_seconds = 0;
-  DBUG_ASSERT(expire_logs_days == 0 || binlog_expire_logs_seconds == 0);
-
-  if (opt_bin_log) {
-    if (expire_logs_days > 0 || binlog_expire_logs_seconds > 0) {
-      time_t purge_time = my_time(0) - binlog_expire_logs_seconds -
-                          expire_logs_days * 24 * 60 * 60;
-      DBUG_EXECUTE_IF("expire_logs_always_at_start",
-                      { purge_time = my_time(0); });
-      mysql_bin_log.purge_logs_before_date(purge_time, true);
-    }
-    if (binlog_space_limit) mysql_bin_log.purge_logs_by_size(true);
-  } else {
-    if (binlog_expire_logs_seconds_supplied)
-      LogErr(WARNING_LEVEL, ER_NEED_LOG_BIN, "--binlog-expire-logs-seconds");
-    if (expire_logs_days_supplied)
-      LogErr(WARNING_LEVEL, ER_NEED_LOG_BIN, "--expire_logs_days");
-    if (binlog_space_limit)
-      LogErr(WARNING_LEVEL, ER_NEED_LOG_BIN, "--binlog-space-limit");
-=======
-  } else if (expire_logs_days_supplied)
-    binlog_expire_logs_seconds = 0;
-  assert(expire_logs_days == 0 || binlog_expire_logs_seconds == 0);
-
-  if (!opt_bin_log) {
-    if (binlog_expire_logs_seconds_supplied)
-      LogErr(WARNING_LEVEL, ER_NEED_LOG_BIN, "--binlog-expire-logs-seconds");
-    if (expire_logs_days_supplied)
-      LogErr(WARNING_LEVEL, ER_NEED_LOG_BIN, "--expire_logs_days");
-    if (binlog_space_limit)
-      LogErr(WARNING_LEVEL, ER_NEED_LOG_BIN, "--binlog-space-limit");
```

Comment:
I suggest to retain upstream's code here and handle `wsrep_recover` after gtid infrastructure initialization.
Please see https://github.com/percona/percona-server/commit/2dc446d46e7ca2309c09ae5bd292a14c52ce3a7e